### PR TITLE
feat: Add Authenticator Checks unit tests

### PR DESCRIPTION
**Pull request by ZeynabRezaei**

### CODE REVIEW

**Comment by navidnabavi** [comment]
Screenshot 2024-03-03 at 12 49 33
Can you cover these branches?

**Comment by ZeynabRezaei** [comment]
@navidnabavi Sure, I will add them.

**Comment by navidnabavi** [comment]
Updates?

**Review by pedy4000 — APPROVED** [review]
🥇 @ZeynabRezaei 

we have to handle "CerberusReasonUpstreamAuthFailed", "CerberusReasonUpstreamAuthTimeout" errors in next steps, and write test for them.

**Comment by pedy4000** [comment]
> Screenshot 2024-03-03 at 12 49 33 Can you cover these branches?

thanks to @ZeynabRezaei 🔥
this tests added to, coverage change from 66.4% to 77.8%.

**Review by pedy4000 — APPROVED** [review]
Welldone 👍🏼

File: pkg/auth/authenticator.go [hunk 1]
```diff
@@ -187,7 +187,7 @@ func (a *Authenticator) Check(ctx context.Context, request *Request) (finalRespo
 		reason, cerberusExtraHeaders = a.TestAccess(request, wsvcCacheEntry)
 
 		extraHeaders = toExtraHeaders(cerberusExtraHeaders)
-		if reason == CerberusReasonOK && hasUpstreamAuth(wsvcCacheEntry) {
+		if reason == "" && hasUpstreamAuth(wsvcCacheEntry) {
```
Comment (navidnabavi) [inline pkg/auth/authenticator.go:190]: why not using constants?

**Pull request merged**